osu.Game.Rulesets.Catch/Difficulty/CatchPerformanceCalculator.cs

```diff
@@ -34,12 +34,10 @@ public override double Calculate(Dictionary<string, double> categoryDifficulty =
         {
             mods = Score.Mods;
 
-            var legacyScore = Score as LegacyScoreInfo;
-
-            fruitsHit = legacyScore?.Count300 ?? Score.Statistics[HitResult.Perfect];
-            ticksHit = legacyScore?.Count100 ?? 0;
-            tinyTicksHit = legacyScore?.Count50 ?? 0;
-            tinyTicksMissed = legacyScore?.CountKatu ?? 0;
+            fruitsHit = Score?.GetCount300() ?? Score.Statistics[HitResult.Perfect];
```

Comment:
`?.` not needed here.
Can be caught later when reviewing nullable reference type.